pkg/envoy/proxy.go

```diff
@@ -130,6 +130,7 @@ func (e *envoy) args(fname string, bootstrapConfig string) []string {
 		// than the flush interval.
 		"--file-flush-interval-msec", "1000",
 		"--disable-hot-restart", // We don't use it, so disable it to simplify Envoy's logic
+		"--allow-unknown-static-fields",
```

Comment:
Why was this changed? Are there unknown fields now that we are accepting?

Reply:
Yeah, because Wasm used a JSON parser that ignored fields with typos so I'm fairly sure some metric overrides had ill-defined fields.

Reply:
Why "static" field flags affects ECDS is still not clear to me, likely a bug somewhere.

Reply:
I am a bit worried we are going to miss real bugs now for this small telemetry case. But at the same time it would not be great if users with broken overrides pod's don't startup when they upgrade.

On the other hand, they would _probably_ immediately find the issue and rollback/fix I would imagine if we did hard fail...

Reply:
We still get a warning for unknown fields.

I think we have to support unknown fields for automatic upgrades, there's just no other way to introduce new fields without putting conditions everywhere.

Reply:
Well this is only for static config, right? So over XDS we are fine; static is tightly coupled with the Envoy version since the bootsrtap is generated by pilot-agent which is in the same container... except now we have bootstrap DS so maybe this makes sense now.

I am a bit confused how this dynamic XDS is impacted by `allow-unknown-static-fields` which I thought was only for startup bootstrap config

Reply:
> I am a bit confused how this dynamic XDS is impacted by `allow-unknown-static-fields` which I thought was only for startup bootstrap config

Me, too, it's probably a bug but it's already shipped so we have to live with it.

Reply:
Fair enough. If its a bug might be worth filing an envoy issue? but this lgtm for now - even with it fixed, for Bootstrap DS we may want it.